app/src/main/java/com/example/stretchy/database/AppDatabase.kt

```diff
@@ -1,28 +1,270 @@
+                        if (it.tId == trainingActivityEntities.getOrNull(indexd + 1)?.tId) {
+                            database.execSQL(
+                                "INSERT INTO training_activities (tId, aId, activityOrder) " +
+                                        "VALUES (${it.tId},$breakId,$index)"
+                            )
+                            index++
+                        } else {
+                            index = 0
+                        }
+                    }
+                }
+
+                fun getTrainingActivitiesList(): List<TrainingActivityEntity> {
+                    val trainingActivityEntities: MutableList<TrainingActivityEntity> =
+                        ArrayList()
+                    val cursor = database.query("SELECT * FROM training_activities")
+
+                    if (cursor.moveToFirst()) {
+                        do {
+                            val tId: Long = cursor.getLong(cursor.getColumnIndex("tId"))
+                            val aId: Long = cursor.getLong(cursor.getColumnIndex("aId"))
+                            val activityOrder: Int =
+                                cursor.getInt(cursor.getColumnIndex("activityOrder"))
+                            val trainingActivityEntity =
+                                TrainingActivityEntity(tId, aId, activityOrder)
+
+                            trainingActivityEntities.add(trainingActivityEntity)
+                        } while (cursor.moveToNext())
+
+                    }
+                    return trainingActivityEntities
+                }
+
+                fun getAllActivities(): List<ActivityEntity> {
+                    val activitiesEntities: MutableList<ActivityEntity> = mutableListOf()
+                    val cursor = database.query("SELECT * from activity")
+                    if (cursor.moveToFirst()) {
+                        do {
+                            val activityId: Long =
+                                cursor.getLong(cursor.getColumnIndex("activityId"))
+                            val name: String = cursor.getString(cursor.getColumnIndex("name"))
+                            val duration: Int = cursor.getInt(cursor.getColumnIndex("duration"))
+                            val activityType: String =
+                                cursor.getString(cursor.getColumnIndex("activityType"))
+                            val activityEntity =
+                                ActivityEntity(
+                                    activityId,
+                                    name,
+                                    duration,
+                                    ActivityType.valueOf(activityType)
+                                )
+
+                            activitiesEntities.add(activityEntity)
+                        } while (cursor.moveToNext())
+                    }
+                    return activitiesEntities
+                }
+
+                fun fixDuplication() {
```

Comment:
`fixDulplication` is too general, maybe lets go with `removeDuplicateActivities`